lldb/source/Symbol/SaveCoreOptions.cpp

```diff
@@ -115,8 +115,21 @@ const MemoryRanges &SaveCoreOptions::GetCoreFileMemoryRanges() const {
   return m_regions_to_save;
 }
 
-Status SaveCoreOptions::EnsureValidConfiguration(
-    lldb::ProcessSP process_sp) const {
+lldb::ThreadCollectionSP SaveCoreOptions::GetThreadsToSave() const {
```

Comment:
Move this down after EnsureValidConfiguaration to avoid merge issues. Return a `ThreadCollection` object by value here. Most internal clients don't need the shared pointer part of it, and the SB API can just copy/move it into a shared pointer.

Reply:
The SBThreadCollection's protected constructor only supports an ThreadCollection_SP. So without changing that I won't be able to only pass by value.

Reply:
Have this function return a ThreadCollection::collection by value. Then you can construct a ThreadCollection from it.